tests/test_url_build.py

```diff
@@ -32,12 +32,16 @@ def test_build_with_scheme_and_host():
     assert u == URL("http://127.0.0.1")
 
 
-def test_build_with_port():
-    with pytest.raises(ValueError):
-        URL.build(port=8000)
-
-    u = URL.build(scheme="http", host="127.0.0.1", port=8000)
-    assert str(u) == "http://127.0.0.1:8000"
+@pytest.mark.parametrize(
+    "port,exc",
+    [
+        pytest.param(8000, ValueError, id="port-only"),
+        pytest.param("", TypeError, id="port-str"),
+    ],
+)
+def test_build_with_port(port, exc):
+    with pytest.raises(exc):
```

Comment:
It is strongly recommended to always use the `match=` regexp with `pytest.raises()`, especially when testing against stdlib exceptions which may come from a lot of places, silently making the test false-positive over time.

Could you integrate this?